src/psi.rs

```diff
@@ -275,3 +281,45 @@ fn hash_2(input1: u64, input2: Fq) -> Fq {
 
     hash
 }
+
+/// given x coords construct Li polynomials
+fn construct_lagrange_basis<F: FftField>(evaluation_domain: &[F]) -> Vec<DensePolynomial<F>> {
+    let mut bases = Vec::with_capacity(evaluation_domain.len());
+    for i in 0..evaluation_domain.len() {
+        let mut l_i = DensePolynomial::from_coefficients_slice(&[F::one()]);
+        let x_i = evaluation_domain[i];
+        for (j, _) in evaluation_domain.iter().enumerate() {
```

Comment:
the mix of 0..len and .enumerate while ignore the value is weird. just be consistent. also in the outer loop, you ARE using the value (`x_i`) so use `.enumerate()`

Reply:
I agree. I copied the code from `fast-eval` but can still change it.